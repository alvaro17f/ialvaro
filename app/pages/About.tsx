import Content from "@/components/Content";
import Form from "@/components/Form";
import Header from "@/components/Header";

export default function About() {
	return (
		<section id="about">
			<Header title="About" />

			<Content>
				<div className="grid md:gap-10 md:grid-cols-[1fr_2fr] place-items-center">
					<div className="mb-5">
						<img
							src="/images/profile.png"
							alt="profile picture"
							width={200}
							height={200}
							className="border-[5px] rounded-full border-azama-white"
						/>
					</div>
					<div className="max-w-[75ch]">
						<h2 className="text-3xl">Hello World!</h2>
						<p>
							My name is √Ålvaro and I am a Full Stack Developer.
							<br />
							I feel very fortunate to have found what I am passionate about and
							to make it my job.
							<br />
							I have also worked managing, training and leading marketing teams
							at a national level.
							<br />
							If there is one thing that characterises me, it is that I am 100%
							involved in everything I do.
							<br />
							I am hard-working, perfectionist and non-conformist.
							<br />
							I have a direct and creative working style. And I put all the
							energy I get from working in what I am so enthusiastic about.
						</p>
					</div>
				</div>
			</Content>

			<Form />
		</section>
	);
}
