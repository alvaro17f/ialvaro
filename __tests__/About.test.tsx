import { expect, it, describe, beforeEach, afterEach } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import About, { metadata } from "@/app/about/page";

describe("<About />", () => {
	beforeEach(() => {
		render(<About />);
	});
	afterEach(cleanup);

	it("should match the snapshot", () => {
		const { container } = render(<About />);
		expect(container).toMatchSnapshot();
	});

	it("should render the <Header />", () => {
		expect(
			screen.getByRole("heading", { level: 1, name: /about/i }),
		).toBeDefined();
	});

	it("should match metadata's title to <Header /> title", () => {
		expect(metadata.title).toStrictEqual("About");
	});

	it("should render the <Content />", () => {
		expect(screen.getByLabelText(/content/i)).toBeDefined();
	});

	it("should render the <Form />", () => {
		expect(screen.getByLabelText(/contact-form/i)).toBeDefined();
	});
});
