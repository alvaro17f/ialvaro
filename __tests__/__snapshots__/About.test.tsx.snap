// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<About /> > should match the snapshot 1`] = `
<div>
  <section
    class="p-5 mb-5 rounded-lg text-azama-dark bg-azama-primary"
    style="opacity: 0; transform: translateX(0px) translateY(0px) scaleY(0.5) translateZ(0);"
  >
    <h1
      class="text-5xl md:text-7xl"
      style="opacity: 0;"
    >
      About
    </h1>
  </section>
  <section
    aria-label="content"
    class="p-5 mb-5 rounded-lg text-azama-dark bg-azama-white"
    style="opacity: 0; transform: translateX(-800px) translateZ(0);"
  >
    <article
      class="p-5 animate__animated animate__fadeIn animate__delay-1.2s"
    >
      <div
        class="grid md:gap-10 md:grid-cols-[1fr_2fr] place-items-center"
      >
        <div
          class="mb-5"
        >
          <img
            alt="profile picture"
            class="border-[5px] rounded-full border-azama-muted"
            height="200"
            src="/images/profile.png"
            width="200"
          />
        </div>
        <div
          class="max-w-[75ch]"
        >
          <h2
            class="text-3xl"
          >
            Hello World!
          </h2>
          <p>
            My name is √Ålvaro and I am a Full Stack Developer.
            <br />
            I feel very fortunate to have found what I am passionate about and to make it my job.
            <br />
            I have also worked managing, training and leading marketing teams at a national level.
            <br />
            If there is one thing that characterises me, it is that I am 100% involved in everything I do.
            <br />
            I am hard-working, perfectionist and non-conformist.
            <br />
            I have a direct and creative working style. And I put all the energy I get from working in what I am so enthusiastic about.
          </p>
        </div>
      </div>
    </article>
  </section>
  <section
    aria-label="section-form"
    class="grid grid-cols-1 md:grid-cols-[1fr_2fr] place-items-center p-5 mb-5 rounded-lg text-azama-dark bg-azama-secondary"
    style="opacity: 0; transform: translateX(-70px) scale(0.7) translateZ(0);"
  >
    <div
      class="mb-5 md:mb-0 md:text-center"
    >
      <h2
        class="text-3xl"
      >
        CONTACT ME!
      </h2>
      <p>
        It's free
      </p>
    </div>
    <form
      aria-label="contact-form"
      class="grid gap-5 p-2 place-items-center md:w-full"
    >
      <input
        aria-label="name"
        class="bg-azama-base placeholder:text-azama-danger text-azama-white max-w-[15rem] md:w-[42rem] md:max-w-none p-5 shadow-md rounded-xl shadow-azama-base"
        defaultvalue=""
        name="name"
        placeholder="Name *"
        required=""
        type="text"
      />
      <input
        aria-label="email"
        class="bg-azama-base placeholder:text-azama-danger text-azama-white max-w-[15rem] md:w-[42rem] md:max-w-none p-5 shadow-md rounded-xl shadow-azama-base"
        defaultvalue=""
        name="email"
        placeholder="Email *"
        required=""
        type="email"
      />
      <input
        aria-label="phone"
        class="bg-azama-base placeholder:text-azama-muted text-azama-white max-w-[15rem] md:w-[42rem] md:max-w-none p-5 shadow-md rounded-xl shadow-azama-base"
        defaultvalue=""
        name="phone"
        placeholder="Phone"
        type="text"
      />
      <textarea
        aria-label="message"
        class="bg-azama-base text-azama-white max-w-[15rem] md:w-[42rem] md:max-w-none p-8 shadow-md rounded-xl shadow-azama-base placeholder:text-azama-danger"
        name="message"
        placeholder="Your message goes here... *"
        required=""
      />
      <button
        aria-label="submit"
        class="p-4 text-5xl rounded-full shadow-md md:w-40 shadow-azama-base bg-azama-base"
        tabindex="0"
        type="submit"
      >
        üî•
      </button>
    </form>
  </section>
</div>
`;
